analyzers/tests/SonarAnalyzer.UnitTest/Helpers/ParameterLoaderTest.cs

```diff
@@ -100,47 +101,81 @@ public void SetParameterValues_WhenGivenValidSonarLintFileAndDoesNotContainAnaly
 
-            // Act with the file on disk
-            ParameterLoader.SetParameterValues(analyzer, options);
-            analyzer.Maximum.Should().Be(1); // from the XML on disk
-
-            // Update the file on disk
-            var originalContent = File.ReadAllText(sonarLintXmlRelativePath);
-            var modifiedContent = originalContent.Replace("<Value>1</Value>", "<Value>42</Value>");
-            File.WriteAllText(sonarLintXmlRelativePath, modifiedContent);
+            // Act
             ParameterLoader.SetParameterValues(analyzer, options);
-            analyzer.Maximum.Should().Be(42);
 
-            // revert change
-            File.WriteAllText(sonarLintXmlRelativePath, originalContent);
+            // Assert
+            analyzer.Maximum.Should().Be(1); // In-memory value
         }
 
         [TestMethod]
-        public void SetParameterValues_WithMalformedXml_DoesNotPopulateProperties()
+        public void SetParameterValues_CalledTwiceAfterChangeInConfigFile_UpdatesProperties()
         {
             // Arrange
-            var options = TestHelper.CreateOptions("ResourceTests\\Malformed\\SonarLint.xml");
+            const string fakeSonarLintXmlFilePath = "ThisPathDoesNotExist\\SonarLint.xml";
+            const string originalSonarLintXmlContent = @"<?xml version=""1.0"" encoding=""UTF-8""?>
+<AnalysisInput>
+  <Settings />
+  <Rules>
+    <Rule>
+      <Key>S1067</Key>
+      <Parameters>
+        <Parameter>
+          <Key>max</Key>
+          <Value>1</Value>
+        </Parameter>
+      </Parameters>
+    </Rule>
+  </Rules>
+  <Files>
+  </Files>
+</AnalysisInput>";
+
+            var options = TestHelper.CreateOptions(fakeSonarLintXmlFilePath, SourceText.From(originalSonarLintXmlContent));
             var analyzer = new ExpressionComplexity(); // Cannot use mock because we use reflection to find properties.
 
             // Act
             ParameterLoader.SetParameterValues(analyzer, options);
+            analyzer.Maximum.Should().Be(1);
 
-            // Assert
-            analyzer.Maximum.Should().Be(3); // Default value
+            // Modify the in-memory additional file
+            var modifiedSonarLintXmlContent = originalSonarLintXmlContent.Replace("<Value>1</Value>", "<Value>42</Value>");
+            var modifiedOptions = TestHelper.CreateOptions(fakeSonarLintXmlFilePath, SourceText.From(modifiedSonarLintXmlContent));
+
+            ParameterLoader.SetParameterValues(analyzer, modifiedOptions);
+            analyzer.Maximum.Should().Be(42);
         }
 
         [TestMethod]
-        public void SetParameterValues_WithBadPath_DoesNotPopulateProperties()
+        public void SetParameterValues_WithMalformedXml_DoesNotPopulateProperties()
         {
             // Arrange
-            var options = TestHelper.CreateOptions("ResourceTests\\ThisPathDoesNotExist\\SonarLint.xml");
+            var options = TestHelper.CreateOptions("ResourceTests\\Malformed\\SonarLint.xml");
```

Comment:
where is the malformed xml? not clear from the test. I expect it's null?

To make it clear, you can make it a DataTestMethod (i.e. parameterized) and pass more "malformed" values like null, empty string and malformed XML to cover all cases

Reply:
It's from "ResourceTests\\Malformed\\SonarLint.xml" -- I didn't change it, that's the way it was.

Reply:
Added use-cases

Reply:
My bad